src/olmo_core/distributed/checkpoint/filesystem.py

```diff
@@ -346,37 +346,39 @@ def read_data(self, plan: dist_cp.LoadPlan, planner: dist_cp.LoadPlanner) -> Fut
                 read_item_content_futures.append(
                     executor.submit(self._get_content_for_read, read_item)
                 )
-            read_item_content_results = []
+
             for f in as_completed(read_item_content_futures):
                 try:
-                    read_item_content_results.append(f.result())
+                    read_item, content = f.result()
```

Comment:
Dirk was pointing out to me last night that the threads could still race ahead and cause an OOM if the operations in this loop are somewhat slow. What are your thoughts?